libtransmission/tr-assert.h

```diff
@@ -13,6 +13,10 @@
 
 [[noreturn]] bool tr_assert_report(char const* file, int line, char const* message_fmt, ...) TR_GNUC_PRINTF(3, 4);
 
+#ifdef __APPLE__
+[[noreturn]] void darwin_assert (char const* file, int line, char const* message);
```

Comment:
Instead of 2 `#ifdef`s and a new function, I'd rather implement the whole `tr_assert_report()` function in tr-assert.mm, thus compiling tr-assert-mm on Mac and tr-assert-cc on non-Mac. Also, `darwin_assert()` is currently declared in a public header but doesn't have the `tr_` prefix.